xdcr/rosmar_xdcr.go

```diff
@@ -123,7 +123,46 @@ func (r *rosmarManager) processEvent(ctx context.Context, event sgbucket.FeedEve
 			return true
 		}
 
-		err = opWithMeta(ctx, col, r.fromBucketSourceID, toCas, toXattrs, event)
+		hlv, mou, body, err := getBodyHLVAndMou(event)
+		if err != nil {
+			base.WarnfCtx(ctx, "Replicating doc %s, could not get body, hlv, and mou: %s", event.Key, err)
+			r.errorCount.Add(1)
+			return false
+		}
+		if hlv != nil && mou != nil {
+			pCAS := base.HexCasToUint64(mou.PreviousCAS)
+			if pCAS <= toCas {
+				r.targetNewerDocs.Add(1)
+				base.TracefCtx(ctx, base.KeyWalrus, "Skipping replicating doc %s, cas %d <= %d", docID, event.Cas, toCas)
+				return true
+			}
+		}
+		if hlv == nil {
+			newHlv := db.NewHybridLogicalVector()
+			hlv = &newHlv
+			err := hlv.AddVersion(db.Version{
+				SourceID: r.fromBucketSourceID,
+				Value:    event.Cas,
+			})
+			if err != nil {
+				base.WarnfCtx(ctx, "Replicating doc %s, could not set hlv.AddVersion: %s", event.Key, err)
+				r.errorCount.Add(1)
+				return false
+			}
+			hlv.CurrentVersionCAS = event.Cas
```

Comment:
Similar to the comment above, if the source mutation was an import, should set cvCAS to pCAS.

Reply:
Since XDCR doesn't handle this case, I'm not going to handle this case for rosmar XDCR.

Reply:
This was evaluating the `_mou` for the case where HLV is not created by import, which will not happen for Sync Gateway 4.x, this could only exist from Sync Gateway 3.2+ clients.

In this case, the import would be echoed between clusters, but only once.